### ISSUE TITLE
FlyBuyCore.configure called more than once

### ISSUE THREAD

**Issue by sangamivivek** [BILDIT-Platform/react-native-bildit-flybuy#55]
**Describe the bug**

Getting 'FlyBuyCore.configure called more than once' error on app start, also the error is inconsistent
versions:
 "react-native": "0.64.2",
 "react-native-bildit-flybuy": "2.1.1",

**Screenshots**

![Screenshot_1692624254](https://github.com/BILDIT-Platform/react-native-bildit-flybuy/assets/63785243/b35dfbbb-2471-4f2c-bdf4-aa4e15d54672)

**Config code snippet on App.tsx**

useAsyncEffect(async () => {
    if (Platform.OS === 'ios') {
      FlyBuy.Core.configure(iosToken);
    } else {
     FlyBuy.Core.configure(androidToken);
    }
    FlyBuy.Pickup.configure();
    }, []);

**Comment by mohanenm** [comment]
@sangamivivek by inconsistent, do you mean it sometimes throws this error and sometimes does not?

**Comment by sangamivivek** [comment]
yes @mohanenm ,while closing and opening the app it sometimes throws this error and sometimes does not

**Comment by mohanenm** [comment]
@addingama What would moving it out of the component do? 

We are still running into this error when force closing and reopening the app.

**Comment by mohanenm** [comment]
The weird thing is that this problem is solved by removing the call to `FlyBuy.Pickup.onPermissionChanged(); `...however, when they remove that call we stop getting location updates.

**Comment by mohanenm** [comment]
@sangamivivek ^

**Comment by sangamivivek** [comment]
@addingama **Index.js**

import React from 'react';
import { AppRegistry } from 'react-native';
import App from './src/App';
import { name as appName } from './app.json';

AppRegistry.registerComponent(appName, () => App);

 **Below is the FlyBuy Config snippet in App.tsx** 

  useAsyncEffect(async () => {
    await fetchRemoteConfig();
    const remotevalue = await getAllRemoteValues();
    QuantumMetric.QuantumMetricLibrary.sendEvent(79, JSON.stringify(remotevalue));
    const iosToken = await RemoteConfig.getRemoteValue('flybuy_token_ios')
    const androidToken = await RemoteConfig.getRemoteValue('flybuy_token_android')
    let versionConfig = await RemoteConfig.getRemoteValue('version_check');
    let flags = await RemoteConfig.getRemoteValue('feature_flags');
    if ('force_update' in flags && flags.force_update)  {
    checkVersion(versionConfig)
    }
    await notifee.createChannel({
      id: 'default',
      name: 'TSC Notifications',
      lights: false,
      vibration: true,
      importance: AndroidImportance.HIGH,
    });
    **if (Platform.OS === 'ios') {
      FlyBuy.Core.configure(iosToken);
    } else {
     FlyBuy.Core.configure(androidToken);
    }
    FlyBuy.Pickup.configure();**
    let basketIdStore = await getItemFromPersistance(BASKET_ID_PERSISTANT_KEY);
    if(basketIdStore==null){
      persistItem(GROUP_ORDER_PERSISTANT_KEY, false);
    }
    if (Platform.OS === 'ios') {
      if (!Settings.get('hasRunBefore')) {
        Settings.set({ hasRunBefore: true });
        clearData();
        persistItem(ALREADY_LAUNCHED, false);
      }else{     
        persistItem(ALREADY_LAUNCHED, true);      
      }
    }
  }, []);

  useEffect(() => {
    AppState.addEventListener("change",handleStateChange);
    let unsub = messaging().onMessage(async (remoteMessage) => {
      if (remoteMessage?.message_source === 'flybuy') {
        FlyBuy.Core.handleRemoteNotification(remoteMessage.data);
      } else if (remoteMessage?.data?.message_source === 'flybuy') {
        FlyBuy.Core.handleRemoteNotification(remoteMessage.data);
      } else {
        let now: Date = new Date();
        const {actiontype, action, payload} = remoteMessage.data ?? {};
        let notificationData: NotifeeNotification = {
          title: remoteMessage.notification?.title ?? '',
          body: remoteMessage.notification?.body ?? payload ?? '',
          android: {
            channelId: 'default',
            pressAction: {id: 'default', mainComponent: 'TSC'},
          },
          data:remoteMessage.data
        };
        if (actiontype && action) {
          notificationData['data'] = {actiontype, action};
        }
        notifee.displayNotification(notificationData);
      }

      if((remoteMessage.notification?.title || remoteMessage.notification?.body) && AppState.currentState === "active"){
      let notificationData: NotifeeNotification = {
        title: remoteMessage.notification?.title,
        body: remoteMessage.notification?.body,
        android: {
          channelId: 'default',
          pressAction: {id: 'default', mainComponent: 'TSC'},
        },
        data:remoteMessage.data
      };
      notifee.displayNotification(notificationData);
      }
    });
    return () => {
      unsub();
      AppState.removeEventListener("change", handleStateChange)
    };
  }, []);

**Comment by mohanenm** [comment]
@addingama any thoughts?

**Comment by mohanenm** [comment]
@addingama Just following up on this once more, not quite sure what is happening here, but think it might be related to the wrapper.

**Comment by sangamivivek** [comment]
Thanks @addingama We will try and let you know if any issues.

**Comment by sangamivivek** [comment]
Hi @addingama and @mohanenm,

We have tested both approaches, testing in release build and calling the configure in index.js outside the react component. Unfortunately the error still persists.

To provide a clear overview of the issue. Initially the same codebase works perfectly with react-native-bildit-flybuy version 1.0.4 , however when upgrading to latest version 2.1.1 iOS works fine but android is throwing an error.

Here are the only Changes made in codebase to update the wrapper:

1. Added in android/app/build.gradle

    implementation ("androidx.appcompat:appcompat:1.3.1") {
        version {
        strictly '1.3.1'
        }
    }

2. Added In Android Manifest.xml

      android:fullBackupContent="false"
      tools:replace="android:allowBackup, android:usesCleartextTraffic, android:fullBackupContent"

3. Added Kotlin version in android /build.gradle

kotlinVersion = "1.6.0"

Please let me know if you require any additional information or if there are other steps we should consider to resolve this issue.

**Comment by mohanenm** [comment]
@addingama I recreated it in the development app. It is only happening to me when I select always authorized permissions though. 

cc: @sangamivivek

**Comment by mohanenm** [comment]
I am only able to recreate it when android location permissions are set to “Allow all the time” and FlyBuy.Pickup.onPermissionChanged(); is called.

`FlyBuy.Pickup.onPermissionChanged()` can be implemented when denying location permissions or allowing while-in-use permissions just fine, but breaks with “Allow all the time”. If I remove`FlyBuy.Pickup.onPermissionChanged()`, allow all the time permissions work fine.  Similar to @sangamivivek, I console.logged core.configure and confirmed it only gets called once.  

Code snippet: 

`const createOrder = () => {

    console.log(Platform.OS)

    FlyBuy.Pickup.onPermissionChanged();

    FlyBuy.Core.Orders.createOrder({

      siteId: SITE_ID,

      pid: NEW_PID.toString(),

      customerInfo: CUSTOMER_INFO,

      pickupType: PickupType.CURBSIDE

    })

      .then((order) => console.log('order', order))

      .catch((err) => console.log(err));

  };`

**Comment by sangamivivek** [comment]
@addingama Thank you for the update. I've installed version 2.1.3, and so far, it's been working well without any crashes. We'll continue to monitor and conduct thorough QA testing. If we happen to come across any issues, we'll let you know. Thanks again!
cc: @mohanenm
